File: 119.杨辉三角-ii.ts
/*
 * @lc app=leetcode.cn id=119 lang=typescript
 *
 * [119] 杨辉三角 II
 */

// @lc code=start
function getRow(rowIndex: number): number[] {
    let thisLine = [1];
    for (let i = 1; i <= rowIndex; i += 1) {
        const lastLine = thisLine;
        thisLine = new Array(i + 1).fill(1);
        for (let j = 1; j < i; j += 1) {
            thisLine[j] = lastLine[j - 1] + lastLine[j];
        }
    }
    return thisLine;
}
// @lc code=end

File: 119.杨辉三角-ii.py
#
# @lc app=leetcode.cn id=119 lang=python3
#
# [119] 杨辉三角 II
#

# @lc code=start
from typing import List


class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        thisLine = [1]
        for i in range(1, rowIndex + 1):
            lastLine = thisLine
            thisLine = [1] * (i + 1)
            for j in range(1, i):
                thisLine[j] = lastLine[j - 1] + lastLine[j]
        return thisLine


# @lc code=end
